pkg/storaged/filesystem/utils.jsx

```diff
@@ -29,61 +29,95 @@
     get_fstab_config_with_client,
     find_children_for_mount_point,
 } from "../utils.js";
+import { parse_subvol_from_options } from "../btrfs/utils.jsx";
 import { StorageLink } from "../storage-controls.jsx";
 
 import { mounting_dialog } from "./mounting-dialog.jsx";
 
 const _ = cockpit.gettext;
 
-export function is_mounted(client, block) {
+/* is mounted with an entry in fstab */
+export function is_mounted(client, block, subvol) {
     const block_fsys = client.blocks_fsys[block.path];
     const mounted_at = block_fsys ? block_fsys.MountPoints : [];
-    const config = block.Configuration.find(c => c[0] == "fstab");
-    if (config && config[1].dir.v) {
-        let dir = decode_filename(config[1].dir.v);
-        if (dir[0] != "/")
-            dir = "/" + dir;
+    const [, dir] = get_fstab_config(block, false, subvol);
+    if (dir) {
         return mounted_at.map(decode_filename).indexOf(dir) >= 0;
     } else
         return null;
 }
 
-export function get_fstab_config(block, also_child_config) {
-    return get_fstab_config_with_client(client, block, also_child_config);
+export function get_fstab_config(block, also_child_config, subvol) {
+    return get_fstab_config_with_client(client, block, also_child_config, subvol);
 }
 
-function find_blocks_for_mount_point(client, mount_point, self) {
-    const blocks = [];
+function nice_block_name(block) {
+    return block_name(client.blocks[block.CryptoBackingDevice] || block);
+}
+
+function find_blocks_for_mount_point(client, mount_point, self_block, self_subvol) {
+    function same_btrfs_volume(a, b) {
+        return (client.blocks_fsys_btrfs[a.path] &&
+                client.blocks_fsys_btrfs[b.path] &&
+                client.blocks_fsys_btrfs[a.path].data.uuid == client.blocks_fsys_btrfs[b.path].data.uuid);
+    }
+
+    function same_btrfs_subvol(a, b) {
+        return (a && b &&
+                ((a.pathname && a.pathname == b.pathname) ||
+                 (a.id && a.id == b.id)));
+    }
+
+    function is_self(b, subvol) {
+        if (self_subvol)
+            return same_btrfs_volume(b, self_block) && same_btrfs_subvol(subvol, self_subvol);
+        else
+            return self_block && b == self_block;
+    }
 
-    function is_self(b) {
-        return self && (b == self || client.blocks[b.CryptoBackingDevice] == self);
+    function fmt_block_and_subvol(block, subvol) {
+        if (subvol)
+            return cockpit.format(_("btrfs subvolume $0 of $1"),
+                                  subvol.pathname || subvol.id,
+                                  block.IdLabel || block.IdUUID);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20240111-120527-6a639276-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)